### PR TITLE
enhancement #511: Refactor Discipline entity

### LINKED ISSUE #511: Refactor Discipline entity
Your task will be to refactor all files, related to **Discipline** entity, such as controller, service, responses, DTOs,  etc.

Your steps can be as following:
1. Check service and repository class for potential issues, misspeling or incorect code style.
2. You need to run unit tests for this service and try to enhance them, also check that all necessary service methods are covered.
3. Check responses and DTOs. As we have a lot of changes some methods can return and take undocumentated information.
4. After all responses and DTOs are checked you can separate documentation of endpoints in controller as made in [this](https://github.com/fictadvisor/fictadvisor/pull/497) pr.

### PR DESCRIPTION
**Pull request by goritpukan**
- Change some methods names in service
- Change some methods names in comtroller
- Refactor swagger documentation in controller 
- Add more validation to DTO's
- Add documentation to DTO's

closes #511

### CODE REVIEW

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 1]
```diff
@@ -97,136 +60,48 @@ export class DisciplineController {
...
   })
-  @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_TEACHERS_GET, GroupByDisciplineGuard)
   @Get('/:disciplineId/teachers')
-  async getAllByDiscipline (
+  async getAllTeachersByDiscipline (
     @Param('disciplineId', DisciplineByIdPipe) disciplineId: string,
     @Query('disciplineType') disciplineType: DisciplineTypeEnum,
   ) {
     const teachers = await this.disciplineService.getTeachers(disciplineId, disciplineType);
     return { teachers };
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:74]: use mapper here, not in service

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 2]
```diff
@@ -97,136 +60,48 @@ export class DisciplineController {
...
-  @ApiParam({
-    name: 'disciplineId',
-    required: true,
-    description: 'Id of a discipline to get',
-  })
   @ApiEndpoint({
     summary: 'Get discipline by id',
+    documentation: disciplineDocumentation.GET_BY_ID,
   })
   @Get(':disciplineId')
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:94]:
```suggestion
  })
  @Get('/:disciplineId')
```

File: fictadvisor-api/src/v2/utils/documentation/discipline/updateById.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  unauthorized: {
+    description: `\n
+    UnauthorizedException:
+      Unauthorized`,
+  },
+  forbidden: {
+    description: `\n
+    NoPermissionException: 
+      You do not have permission to perform this action`,
+  },
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/utils/documentation/discipline/updateById.ts:26]: use default resqponses

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 4]
```diff
@@ -40,23 +31,10 @@ export class DisciplineController {
...
-    NoPermissionException:
-      You do not have permission to perform this action`,
-  })
   @Post()
   @ApiEndpoint({
     summary: 'Create new discipline',
+    documentation: disciplineDocumentation.CREATE,
     permissions: PERMISSION.GROUPS_$GROUPID_DISCIPLINES_CREATE,
   })
   async create (
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:40]: add mehod return type

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 5]
```diff
@@ -97,136 +60,48 @@ export class DisciplineController {
...
   @ApiEndpoint({
     summary: 'Get discipline by id',
+    documentation: disciplineDocumentation.GET_BY_ID,
   })
   @Get(':disciplineId')
-  async getById (@Param('disciplineId', DisciplineByIdPipe) disciplineId: string) {
-    const discipline = await this.disciplineService.get(disciplineId);
+  async getById (
+    @Param('disciplineId', DisciplineByIdPipe) disciplineId: string
+  ) {
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:97]: add mehod return type

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts [hunk 6]
```diff
@@ -97,136 +60,48 @@ export class DisciplineController {
...
-    name: 'disciplineId',
-    required: true,
-    description: 'Id of a discipline to update',
-  })
   @ApiEndpoint({
     summary: 'Update discipline by id',
+    documentation: disciplineDocumentation.UPDATE_BY_ID,
     permissions: PERMISSION.DISCIPLINE_UPDATE,
   })
   @Patch('/:disciplineId')
```
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:107]: add mehod return type

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts
Comment (bohdanbulakh) [inline fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:1]: go to `DisciplineMapper`:
- add return types for each method
- rename all mapper methods, they must match such pattern: if the method returns object of type `{SomeType}Response` its name must be `get{SomeType}`

**Review by bohdanbulakh — CHANGES_REQUESTED** [review]

File: fictadvisor-api/src/v2/api/controllers/DisciplineController.ts
Comment (goritpukan) [reply fictadvisor-api/src/v2/api/controllers/DisciplineController.ts:1]: if it returns array, but type name is something like BaseDisciplineResponse[], should i name method getBaseDisciplines or getBaseDiscipline?

**Review by bohdanbulakh — CHANGES_REQUESTED** [review]

**Review by bohdanbulakh — APPROVED** [review]

**Pull request merged**